src/graph/graph.c

```diff
@@ -1110,29 +1126,31 @@ void Graph_DeleteEdges
 		int         r         =  Edge_GetRelationID(e);
 		NodeID      src_id    =  Edge_GetSrcNodeID(e);
 		NodeID      dest_id   =  Edge_GetDestNodeID(e);
+		EdgeID      edge_id   =  ENTITY_GET_ID(e);
 
 		ASSERT(!DataBlock_ItemIsDeleted((void *)e->attributes));
 
 		// an edge of type r has just been deleted, update statistics
 		GraphStatistics_DecEdgeCount(&g->stats, r, 1);
 
-		R = Graph_GetRelationMatrix(g, r, false);
+		M = Graph_GetRelationMatrix(g, r, false);
 
-		// single edge of type R connecting src to dest, delete entry
-		info = RG_Matrix_removeEntry_UINT64(R, src_id, dest_id, ENTITY_GET_ID(e), &entry_deleted);
+		GrB_Index me_id;
+		info = Delta_Matrix_extractElement_UINT64(&me_id, M, src_id, dest_id);
 		ASSERT(info == GrB_SUCCESS);
 
-		if(entry_deleted) {
-			// TODO: consider making ADJ UINT64_T where ADJ[i,j] = #connections
-			// drop the entry once it reaches 0
-			//
+		if(SINGLE_EDGE(me_id)) {
+			info = Delta_Matrix_removeElement(M, src_id, dest_id);
+			ASSERT(info == GrB_SUCCESS);
+			ASSERT(me_id == edge_id);
+
 			// see if source is connected to destination with additional edges
 			bool connected = false;
 			int relationCount = Graph_RelationTypeCount(g);
-			for(int i = 0; i < relationCount; i++) {
-				if(i == r) continue;
-				M = Graph_GetRelationMatrix(g, i, false);
-				info = RG_Matrix_extractElement_UINT64(&x, M, src_id, dest_id);
+			for(int j = 0; j < relationCount; j++) {
```

Comment:
I think that it would be better if we change our `ADJ` matrix from boolean to `UINT` where `ADJ[i,j] = k` `k` being the number of edges connecting `i` to `j` when the value reaches 0 we can delete the entry. this will save this search and we could keep using the ADJ as is.